### PR TITLE
COSBETA-455: Submit minimal information about a non-standard nanomaterial

### PR DESCRIPTION
**Pull request by aidayounesi**
- Nanomaterials dashboard page
- Nanomaterial wizard pages
- Check your answers page
- Confirmation page

## Checklist:

- [ ] Automated checks are passing locally.
- [ ] CHANGELOG updated if change is worth telling users about.
### General testing
- [ ] Test without javascript
- [ ] Test on small screen
### Accessibility testing
- [ ] Works keyboard only
- [ ] Tested with one screen reader
- [ ] Zoom page to 400% - content still visible
- [ ] Disable css - does content make sense and appear in a logical order?
- [ ] Passes automated checker (automated in build or manual)

### CODE REVIEW

File: cosmetics-web/app/views/non_standard_nanomaterials/_submit_non_standard_nanomaterial.html.slim [hunk 1]
```diff
@@ -0,0 +1,12 @@
+.govuk-grid-column-two-thirds
```
Comment (tom-bell-softwire) [inline cosmetics-web/app/views/non_standard_nanomaterials/_submit_non_standard_nanomaterial.html.slim]: I think it's slightly clearer to call this partial `_submit_notification.html.slim`

File: cosmetics-web/app/views/responsible_persons/non_standard_nanomaterials/edit.html.slim [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+- content_for :after_header do
+  = link_to "Back", responsible_person_non_standard_nanomaterials_path(@responsible_person),
+          class: "govuk-back-link"
+.govuk-grid-row
+  .govuk-grid-column-two-thirds
+    h1.govuk-heading-xl
+      | Nanomaterial
+
+  .govuk-grid-column-full
+    = render "non_standard_nanomaterials/non_standard_nanomaterial_details",
```
Comment (tom-bell-softwire) [inline cosmetics-web/app/views/responsible_persons/non_standard_nanomaterials/edit.html.slim]: This partial name feels a bit redundant. Would `_details.html.slim` be clear enough?

File: cosmetics-web/app/views/non_standard_nanomaterial_build/add_iupac_name.html.slim [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  = link_to "Back", responsible_person_non_standard_nanomaterials_path(@responsible_person), class: "govuk-back-link"
+
+= form_with model: @non_standard_nanomaterial, url: wizard_path, method: :put do |form|
+  .govuk-grid-row
+    .govuk-grid-column-two-thirds
+      = render "form_components/govuk_error_summary", form: form
+      h1.govuk-heading-xl
+        = title
+      = render "form_components/govuk_input", key: :iupac_name, form: form,
+              label: { text: question, classes: "govuk-label--m", isPageHeading: false }
```
[the remark this answers is not in the document]
Comment (aidayounesi) [reply cosmetics-web/app/views/non_standard_nanomaterial_build/add_iupac_name.html.slim]: You are right, no need to have it. Removed.

File: cosmetics-web/app/views/responsible_persons/non_standard_nanomaterials/confirm.slim [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+- content_for :page_title, title
+
+.govuk-grid-row
+  .govuk-grid-column-two-thirds
+    h1.govuk-heading-xl
+      = title
+    / TODO: add content here
+    p = link_to "Back to my nanomaterials",
+            responsible_person_non_standard_nanomaterials_path(@responsible_person),
+            class: "govuk-link"
```
[the remark this answers is not in the document]
Comment (aidayounesi) [reply cosmetics-web/app/views/responsible_persons/non_standard_nanomaterials/confirm.slim]: Thanks. Sure, changed.

File: cosmetics-web/spec/controllers/non_standard_nanomaterial_build_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    sign_out
+  end
+
+  describe "GET #new" do
+    it "redirects to the first step of the wizard" do
+      get(:new, params: params)
+      expect(response).to redirect_to(responsible_person_non_standard_nanomaterial_build_path(responsible_person, non_standard_nanomaterial, :add_iupac_name))
+    end
+
+    it "does not allow the user to create a non standard nanomaterial for a Responsible Person they not belong to" do
```
[the remark this answers is not in the document]
Comment (aidayounesi) [reply cosmetics-web/spec/controllers/non_standard_nanomaterial_build_controller_spec.rb]: Sure thanks

File: cosmetics-web/spec/controllers/non_standard_nanomaterial_build_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    end
+  end
+
+  describe "POST #update" do
+    it "assigns the correct non standard nanomaterial" do
+      post(:update, params: params.merge(id: :add_iupac_name, non_standard_nanomaterial: { iupac_name: "4-ethyl" }))
+      expect(assigns(:non_standard_nanomaterial)).to eq(non_standard_nanomaterial)
+    end
+
+    it "does not allow the user to update a non standard nanomaterial that has already been submitted" do
```
[the remark this answers is not in the document]
Comment (aidayounesi) [reply cosmetics-web/spec/controllers/non_standard_nanomaterial_build_controller_spec.rb]: thanks

File: cosmetics-web/spec/system/non_standard_nanomaterial_entry_spec.rb [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+    # add_iupac_name
+    fill_in :non_standard_nanomaterial_iupac_name, with: "A nanomaterial"
+    click_button "Continue"
+
+    # Check your answers page
+    expect_check_your_answer(get_non_standard_nanomaterial_table, "Name", "A nanomaterial")
+    click_button "Accept and submit the nanomaterial"
+
+    # Confirmation page
```
[the remark this answers is not in the document]
Comment (aidayounesi) [reply cosmetics-web/spec/system/non_standard_nanomaterial_entry_spec.rb]: Right now we don't have anything specific to assert, probably later we should assert the confirmation reference number.

File: cosmetics-web/app/controllers/non_standard_nanomaterial_build_controller.rb [hunk 8]
```diff
@@ -0,0 +1,48 @@
+class NonStandardNanomaterialBuildController < ApplicationController
+  include Wicked::Wizard
+
+  steps :add_iupac_name
+
+  before_action :set_responsible_person
+  before_action :set_non_standard_nanomaterial
```
[the remark this answers is not in the document]
Comment (aidayounesi) [reply cosmetics-web/app/controllers/non_standard_nanomaterial_build_controller.rb]: Sure, thanks

File: cosmetics-web/spec/factories/non_standard_nanomaterial.rb [hunk 9]
```diff
@@ -0,0 +1,5 @@
+FactoryBot.define do
+  factory :non_standard_nanomaterial do
+    iupac_name { "4-ethyl" }
```
[the remark this answers is not in the document]
Comment (aidayounesi) [reply cosmetics-web/spec/factories/non_standard_nanomaterial.rb]: What do you think about "butane"?
https://www.siyavula.com/read/science/grade-12/organic-molecules/04-organic-molecules-03

File: cosmetics-web/spec/system/non_standard_nanomaterial_entry_spec.rb (hunk 7 above)
[the remark this answers is not in the document]
Comment (tom-bell-softwire) [reply cosmetics-web/spec/system/non_standard_nanomaterial_entry_spec.rb]: Makes sense!

File: cosmetics-web/app/views/layouts/_navbar.html.slim [hunk 10]
```diff
@@ -4,6 +4,11 @@ nav.app-navigation.govuk-clearfix
       = link_to "Cosmetic products", responsible_person_notifications_path(@responsible_person),
               "data-topnav": "responsible_persons/notifications",
               class: "govuk-link--no-visited-state"
+    li[
+    class=(params[:controller] == "responsible_persons/non_standard_nanomaterials" && "app-navigation--current-page")]
+      = link_to "Nanomaterials", responsible_person_non_standard_nanomaterials_path(@responsible_person),
+              "data-topnav": "responsible_persons/non_standard_nanomaterials",
+              class: "govuk-link--no-visited-state"
```
[the remark this answers is not in the document]
Comment (aidayounesi) [reply cosmetics-web/app/views/layouts/_navbar.html.slim]: It doesn't look perfect, specially on phone. I showed it to Ralph and he said he is going to think about it. I think maybe another ticket for making the navbar more responsive?

File: cosmetics-web/app/views/responsible_persons/non_standard_nanomaterials/index.html.slim [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+
+- content_for :page_title, title
+- content_for :after_header do
+  = render "layouts/navbar"
+.govuk-grid-row
+  .govuk-grid-column-two-thirds
+    h1.govuk-heading-xl
+      = title
+    p
+      | You must submit a notification about any nanomaterials used in #{@responsible_person.name}’s cosmetics products
```
[the remark this answers is not in the document]
Comment (aidayounesi) [reply cosmetics-web/app/views/responsible_persons/non_standard_nanomaterials/index.html.slim]: Sure thanks

File: cosmetics-web/app/views/responsible_persons/non_standard_nanomaterials/confirm.slim [hunk 12]
```diff
@@ -0,0 +1,12 @@
+- title = "Confirmation page"
+
+- content_for :page_title, title
+
+.govuk-grid-row
+  .govuk-grid-column-two-thirds
+    h1.govuk-heading-xl
+      = title
+    / TODO: add content here
```
[the remark this answers is not in the document]
Comment (aidayounesi) [reply cosmetics-web/app/views/responsible_persons/non_standard_nanomaterials/confirm.slim]: After a quick chat with Ralph, it's going to point to the same place that the very last link on the landing page points to. So, covered by COSBETA-219 (?) and points to "/notify-nanomaterials"

File: cosmetics-web/spec/factories/non_standard_nanomaterial.rb (hunk 9 above)
[the remark this answers is not in the document]
Comment (aidayounesi) [reply cosmetics-web/spec/factories/non_standard_nanomaterial.rb]: Toby suggested:  Zinc oxide

File: cosmetics-web/app/views/non_standard_nanomaterials/_non_standard_nanomaterial_details.html.slim [hunk 13]
```diff
@@ -0,0 +1,7 @@
+table.govuk-table.check-your-answers-table#non-standard-nanomaterial-table
```
[the remark this answers is not in the document]
Comment (aidayounesi) [reply cosmetics-web/app/views/non_standard_nanomaterials/_non_standard_nanomaterial_details.html.slim]: As discussed not part of this PR anymore.

File: cosmetics-web/app/views/responsible_persons/non_standard_nanomaterials/edit.html.slim [hunk 14]
```diff
@@ -0,0 +1,16 @@
+- content_for :page_title, "Check your answers"
+- content_for :after_header do
+  = link_to "Back", responsible_person_non_standard_nanomaterials_path(@responsible_person),
+          class: "govuk-back-link"
+.govuk-grid-row
+  .govuk-grid-column-two-thirds
+    h1.govuk-heading-xl
```
[the remark this answers is not in the document]
Comment (aidayounesi) [reply cosmetics-web/app/views/responsible_persons/non_standard_nanomaterials/edit.html.slim]: As discussed not part of this PR anymore.

**Pull request merged**